tensorflow/java/src/test/java/org/tensorflow/DirectSessionTest.java

```diff
@@ -0,0 +1,134 @@
+/* Copyright 2020 Elphel, Inc.
+
+SPDX-License-Identifier: Apache-2.0
+==============================================================================*/
+
+package org.tensorflow;
+
+import static org.junit.Assert.assertEquals;
+
+import java.nio.FloatBuffer;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class DirectSessionTest {
+  private static final float EPSILON_F = 1e-7f;
+
+  @Test
+  public void DirectSessionTest() {
+    try (Graph g = new Graph();
+        Session s = new Session(g)) {
+
+      Operation input1 = g.opBuilder("Placeholder","input1")
+                          .setAttr("dtype",DataType.FLOAT)
+                          .build();
+
+      Operation input2 = g.opBuilder("Const", "input2")
+                          .setAttr("dtype", DataType.FLOAT)
+                          .setAttr("value", Tensor.<Float>create((float) 2.0, Float.class))
+                          .build();
+
+      Operation output = g.opBuilder("Mul", "output1")
+                          .addInput(input1.output(0))
+                          .addInput(input2.output(0))
+                          .build();
+
+      int size = 256;
+
+      long[] shape = {size};
+      float[] floats = new float[size];
+      float[] expected = new float[size];
+
+      for (int i = 0; i < size; ++i) {
+        floats[i] = i+1;
+        expected[i] = 2*floats[i];
+      }
+
+      Tensor<Float> t_cpu = Tensor.create(shape,FloatBuffer.wrap(floats));
+
+      // dummy run
+      Tensor<?> res_cpu = s.runner().fetch("output1").feed("input1", t_cpu).run().get(0);
+
+      // don't have to test a regular run
+      /*
+      float [] res1 = new float[floats.length];
+      res_cpu.copyTo(res1);
+
+      for (int i = 0; i < size; ++i) {
+        assertEquals(expected[i], res1[i], EPSILON_F);
+      }
+      */
+
+      //String gpuDeviceName = s.GPUDeviceName();
+      // Expecting but not insisting on string "/job:localhost/replica:0/task:0/device:GPU:0"
+      //byte[] gpuName = gpuDeviceName.getBytes(UTF_8);
+
+      /*
+       * The following message is copied from SavedModelBundleTest.java:
+       */
+
+      // Ideally this would use the generated Java sources for protocol buffers
+      // and end up with something like the snippet below. However, generating
+      // the Java files for the .proto files in tensorflow/core:protos_all is
+      // a bit cumbersome in bazel until the proto_library rule is setup.
+      //
+      // See https://github.com/bazelbuild/bazel/issues/52#issuecomment-194341866
+      // https://github.com/bazelbuild/rules_go/pull/121#issuecomment-251515362
+      // https://github.com/bazelbuild/rules_go/pull/121#issuecomment-251692558
+      //
```

Comment:
Just a note that in the new generation of [TF Java](https://github.com/tensorflow/java), this is not a problem anymore, as proto definitions are directly available from the Java client.